libs/sdk-ui-filters/src/DateFilter/DynamicSelect/DynamicSelect.tsx

```diff
@@ -145,6 +153,23 @@ export class DynamicSelect extends React.Component<IDynamicSelectProps, IDynamic
                         visibleItemsRange,
                     };
 
+                    const onBlur = (event: React.FocusEvent<HTMLInputElement>): void => {
+                        const currentValue = (event.target as HTMLInputElement).value;
+                        if (resetOnBlur) {
+                            selectItem(selectedItem);
+                            this.onInputValueChanged(selectedItem ? selectedItem.label : "");
+                        } else if (customValueValidator(currentValue)) {
+                            selectItem({
+                                type: "option",
+                                value: Number(currentValue),
+                                label: currentValue,
+                            });
+                            this.onInputValueChanged(currentValue);
+                        } else {
+                            this.onInputValueChanged(value.toString());
+                        }
+                    };
+
```

Comment:
I know this is not your pate. But can we make this component little bit more clear? And move it outside of render body of it's parent? ... It's horrible

Reply:
I will move it but it does not look much better thanks to the dependencies from the render method.

Reply:
Fixed.